src/types/on_disk/manifest_file.rs

```diff
@@ -185,8 +215,33 @@ impl TryFrom<DataFile> for types::DataFile {
     }
 }
 
+impl TryFrom<types::DataFile> for DataFile {
+    type Error = Error;
+
+    fn try_from(v: types::DataFile) -> Result<DataFile> {
```

Comment:
I prefer to have a `format_data_file` function which won't leak the convertion function at `types::DataFile` level.